crates/uv/src/commands/project/init.rs

```diff
@@ -589,11 +602,25 @@ impl InitProjectKind {
         requires_python: &RequiresPython,
         python_request: Option<&PythonRequest>,
         vcs: Option<VersionControlSystem>,
+        no_authors: bool,
         no_readme: bool,
         package: bool,
     ) -> Result<()> {
+        fs_err::create_dir_all(path)?;
+
+        // Get the author information from git configuration.
+        let author = if no_authors {
```

Comment:
Hi! For non-packaged application project types, I would probably end up deleting the author info most of the time.

What do you think about only adding it for libraries + package=true app types? ie: Putting this block inside an `if package` conditional?

Reply:
That seems reasonable to me, I think.

Reply:
Regarding `--authors`, shoule we enable it to accept a value? For example: `uv init --authors 'Alice <alice@exmaple.org>'` or `uv init --authors Alice`.

Reply:
That's... a good question. It makes sense to allow providing the authors, but also it breaks our typical `--no-<flag>` / `--<flag>` pattern. I think we'd need like... `--authors git` as a special value that says fetch from git?

Reply:
or rename this flag as something like `--fill-authors` / `--no-fill-authors` and keep `--authors` for future usage?

Reply:
I'm hesitant to introduce a bunch of flags for this. I'll do some thinking out loud. Typically, we use singular names for options that can be repeated so... `--author <name> --author <another name>` makes sense to me. Then... `--no-authors` with a `--no-author` alias to disable detection? Then we're left with "I want the author to be derived from somewhere even if it's an app" which could be like `--force-author` or we can reuse `--author` and add a special key like `auto`. This is sort of nice because we can add more special keys like `git`, `os`, etc. for various author sources? This would let you force read from a source e.g. `uv init --vcs git --author os` which would otherwise be another new flag like `--author-from`? I think collisions are pretty unlikely but it does make the flag less self-documenting.

I shall seek another opinion here. cc @BurntSushi

Reply:
I guess my initial reaction here is that it's not worth changing this flag's behavior based on library versus application. It's not immediately obvious to me why we would make that distinction by default. If folks don't want author information, then instead of deleting it after-the-fact, they can do `--no-authors`.

Otherwise, how far do we want to go with `uv init` being an interface for writing metadata values to `pyproject.toml`? If we just want it to be something that tries to infer things with reasonable defaults, then we might not need to expose functionality like `--author BurntSushi`.

Failing all of that, I like your approach @zanieb. I might use `infer-git` instead of just `git` for example. Or something like that.

Reply:
> I guess my initial reaction here is that it's not worth changing this flag's behavior based on library versus application. It's not immediately obvious to me why we would make that distinction by default

I guess I like sensible defaults. I don't think the default has much bearing on the flag name though, we still need both toggles for consistency with the rest of our CLI.

Right now we don't really support passing metadata directly, but I think we generally should in the future so I want to design in a way that we won't regret.

Reply:
> I guess my initial reaction here is that it's not worth changing this flag's behavior based on library versus application. It's not immediately obvious to me why we would make that distinction by default. If folks don't want author information, then instead of deleting it after-the-fact, they can do `--no-authors`.

For me, the reason why I wouldn't want authors (or any other optional fields related to packaging) defined in a `pyproject.toml` for an app project type, is that if the package doesn't have a build backend and isn't being published to PyPI, then it's extra clutter/something to understand/keep up to date that isn't adding any value. (So really this is more about build backend/publishing vs not, rather than app vs library.)

Users could of course delete those fields afterwards, but they'd have to know that they were optional or else be willing to try deleting them and seeing if something breaks (and hoping that the tool's schema validation is solid enough that it gives me a suitable upfront error message rather than something silently breaking later).

Would a compromise that could keep `uv init` similar (and avoid the need to support multiple VCSes and OS username lookup etc) be to instead not populate authors at all in `uv init` and instead have `uv publish` handle making the package "publish ready" wrt best practices etc (eg prompting if metadata not filled out).